script.js

```diff
@@ -1,4 +1,569 @@
+    numDiceChosen = Number(input);
+
+    // If this is the second (or more) round, skip submode selection and go straight into playing the game
+    if (subMode == SUB_MODE_REGULAR) {
+      gameMode = GAME_MODE_DICE_ROLL;
+    } else if (subMode == SUB_MODE_KNOCKOUT) {
+      gameMode = GAME_MODE_SELECT_OPPONENTS;
+      return (
+        'You have chosen to play the next game with ' +
+        numDiceChosen +
+        " dice. Click submit to see who you'll be going up against. "
+      );
+    } else {
+      // Else this must be the first round; prompt user to chose a submode
+      // Change to dice roll mode
+      gameMode = GAME_MODE_CHOOSE_SUB_MODE;
+      return (
+        'You have chosen to play with ' +
+        numDiceChosen +
+        ' dice. <br><br> Please enter one of the following game types: <br>1. regular <br>2. knockout '
+      );
+    }
+  }
+
+  if (gameMode == GAME_MODE_CHOOSE_SUB_MODE) {
+    // valdiate the user input
+    if (input != SUB_MODE_REGULAR && input != SUB_MODE_KNOCKOUT) {
+      return 'Please enter one of the following game types: <br>1. regular <br>2. knockout ';
+    }
+
+    // If user wants to play a regular game, change the mode and prompt player 1 to start
+    if (input == SUB_MODE_REGULAR) {
+      // set the submode
+      subMode = SUB_MODE_REGULAR;
+
+      // Since all players in playerProfiles are playing, push each player into currRoundParticpants
+      for (var i = 0; i < playerProfiles.length; i += 1) {
+        currRoundParticipants.push(playerProfiles[i]);
+      }
+      // Progress the game to the next mode
+      gameMode = GAME_MODE_DICE_ROLL;
+      // Prompt Player 1 to roll his dice
+      return (
+        'You have selected to play a ' +
+        subMode +
+        ' game. Player 1, please click submit to roll your dice'
+      );
+    }
+    // If user does not want to play a regular game, then he wants to play knockout. Change the game mode to programitically select opponenets
+    if (input == SUB_MODE_KNOCKOUT) {
+      // set the submode
+      subMode = SUB_MODE_KNOCKOUT;
+      // Progress the game to the next mode
+      gameMode = GAME_MODE_SELECT_OPPONENTS;
+    }
+  }
+
+  //Randomly get 2 players to play against each other
+  if (gameMode == GAME_MODE_SELECT_OPPONENTS) {
+    // If we're in the first round, currPlayers arr would be empty
+    if (currRoundParticipants.length < 1) {
+      // set currRoundParticpants to include any 2 random players from the playerProfiles array
+      currRoundParticipants = [getPlayerRandomly(), getPlayerRandomly()];
+    }
+    // Else, we are in this mode because a player has won a previous round. add one other player to currRoundParticipants
+    else {
+      currRoundParticipants.push(getPlayerRandomly());
+    }
+    // Progress the game by changing to the next mode
+    gameMode = GAME_MODE_DICE_ROLL;
+
+    // Prompt users on who will be playing in this round
```

Comment:
Where is this comment attached to? Seems dangling. If attached to the code below, no need newline between comment and code